src/sa.py

```diff
@@ -1,15 +1,149 @@
 import argparse
+import fasta, fastq
+import re
 
+def getSuffixes(x: memoryview):
+    """
+    Gets all suffixes from a string
+    """
+    suffixes = [x[i:] for i in range(0, len(x))] 
+    return suffixes
+
+def getTrailingNumber(s):
+    m = re.search(r'\d+$', s)
+    return (s[:m.start()], int(s[m.start():]))
+
+def strcmp(s1, index, x):
+    max_i = min(len(s1), len(x)-index)
+    i = 0
+    while i < max_i and s1[i] == x[index+i]:
+        i += 1
+        if i >= max_i:
+            return 0
+    if s1[i] > x[index+i]: return 1
+    else: return -1
+
+
+def search(sa, pattern, genome):
+    low = 0
+    high = len(sa)-1
```

Comment:
Gutsy to include the end index in the interval when all the algorithms we have used elsewhere exclude it, but it can work, of course.